vehicle_sim/src/sim.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/python3
+
+from datetime import datetime, timezone
+import math
+import numpy as np
+import rospy
+import tf
+import utm
+
+from geometry_msgs.msg import PoseStamped, TwistStamped
+from geometry_utils.geometry_utils import compute_steering_angle_from_curvature, State
+from nmea_msgs.msg import Gprmc
+from std_msgs.msg import Float64
+from vehicle_sim import LateralModel, LongitudinalModel, VehicleSim
+
+
+class Sim:
+    def __init__(self):
+
+        # Constants
+        rate = rospy.get_param("~rate", 10.0)
+        self.time_step_secs = rospy.get_param(
+            "~time_step_secs", 0.1
+        )  # Ideally calculate that from rate above
+        desired_acc_gain_p = rospy.get_param("~desired_acc_gain_p", 0.5)
+        wheelbase = rospy.get_param("~wheelbase", 0.406)
+        max_radius = rospy.get_param("~max_radius", 1000)
+        self.mps_to_knots = 1.94384
+
+        # Subscribers
+        rospy.Subscriber(
+            "initialpose",
+            PoseStamped,
+            self.clicked_point_callback,
+        )
+        rospy.Subscriber(
+            "target_velocity",
+            TwistStamped,
+            self.target_velocity_callback,
+        )
+        rospy.Subscriber(
+            "target_curvature",
+            Float64,
+            self.target_curvature_callback,
+        )
+
+        # Publishers
+        self.sim_pose_pub = rospy.Publisher(
+            "current_pose_sim",
+            PoseStamped,
+            queue_size=10,
+        )
+        self.sim_twist_pub = rospy.Publisher(
+            "current_velocity_sim",
+            TwistStamped,
+            queue_size=10,
+        )
+        self.rmc_pub = rospy.Publisher("gps_info", Gprmc, queue_size=10)
+
+        # Initialization
+        self.simulated_vehicle = VehicleSim(
+            lateral_model=LateralModel.PERFECT,
+            longitudinal_model=LongitudinalModel.ACC_P_CONTROl,
+            wheelbase_m=wheelbase,
+            desired_acc_gain_p=desired_acc_gain_p,
+        )
+
+        self.state_initialized = False
+        self.target_curvature = float(1 / max_radius)
+        self.target_velocity = 0
+        self.listener = tf.TransformListener()
```

Comment:
Renaming suggestion:
```suggestion
        self.tf_listener = tf.TransformListener()
```

Reply:
Done